vowpalwabbit/parse_regressor.cc

```diff
@@ -377,30 +379,30 @@ void save_load_header(
       bytes_read_write += bin_text_read_write_fixed_validated(model_file, nullptr, 0, "", read, msg, text);
 
       // TODO: validate skips?
-      uint32_t skip_len = (uint32_t)all.skip_strings.size();
+      uint32_t skip_len = (g_transformer != nullptr) ? g_transformer->get_initial_skip_definitions().size() : 0;
       msg << skip_len << " skip:";
       bytes_read_write +=
           bin_text_read_write_fixed_validated(model_file, (char*)&skip_len, sizeof(skip_len), "", read, msg, text);
 
+      const auto& skip_strings = g_transformer != nullptr ? g_transformer->get_initial_skip_definitions() : temp_vec;
       for (size_t i = 0; i < skip_len; i++)
       {
         char skip[4] = {0, 0, 0, 0};
         if (!read)
         {
-          msg << all.skip_strings[i] << " ";
-          memcpy(skip, all.skip_strings[i].c_str(), std::min(static_cast<size_t>(3), all.skip_strings[i].size()));
+          msg << skip_strings[i] << " ";
+          memcpy(skip, skip_strings[i].c_str(), std::min(static_cast<size_t>(3), skip_strings[i].size()));
         }
 
         bytes_read_write += bin_text_read_write_fixed_validated(model_file, skip, 3, "", read, msg, text);
         if (read)
         {
           std::string temp(skip);
```

Comment:
I removed `all.skip_strings.push_back(temp);` because reading these in is already handled by the `file_options` built below